src/common/reducers/repositories.js

```diff
@@ -1,42 +1,117 @@
+import merge from 'lodash/merge';
+import union from 'lodash/union';
+
 import * as ActionTypes from '../actions/repositories';
-import { parseGitHubRepoUrl } from '../helpers/github-url';
+
+// XXX move to entities.js
+// XXX should we be handling snaps here too?
+// no, because it's not part of the normalized result from github
+// so this should really be called github_entities
+// no, decompose instead ...
+export function entities(state = { snaps: {}, repos: {} }, action) {
+  if (action.payload && action.payload.entities) {
+    return merge({}, state, action.payload.entities);
+  }
+
+
+  const reposInitialState = {
+    __isSelected: false,
```

Comment:
I'm not big fan of `__` prefix as it's going to make usages of this look like `entitites.repos[is].__isSelected` which is a bit confusing - why I need to remember which properties are custom with `__` and which are not?

Similar issue is with camelCase vs snake_case in names. Object returned by GH and LP APIs are snake_cased while the propetries we use (and add here to the same objects) are camelCased. So it ends up like:`entities.repo[i].full_name` but `entities.repo[i].__isSelected`.

But I'm not sure how to make those API and internal properties compatible in notation but without possible clobbering - apart from additional nesting which is not perfect either.